### PR TITLE
Pick up curl 7.72.0 current

### PR DESCRIPTION
**Pull request by wrowe**
Commit Message: Pick up curl 7.72.0 current

Additional Description:

- Picks up fix of CVE-2020-8231: libcurl: wrong connect-only connection
  (Envoy does NOT use the affected CURLOPT_CONNECT_ONLY toggle,
  this change simply satisfies overly-simplistic audits, and this further 
  reinforces issue https://github.com/envoyproxy/envoy/issues/11816 )

- Resolve premature force-AF_UNIX toggle in our working branch and
  disables UNIX_SOCKETS on windows, until afunix.h is used for sys/un.h 
  (Breakage is between curl 7.69 and curl 7.72, which doesn't pick up afunix.h 
  on windows; will address and resolve upstream before revisiting at envoy.)

- Work around misplaced -MD cflag on Windows build
  See https://github.com/bazelbuild/rules_foreign_cc/issues/426
  for comprehensive discussion of the defect. This workaround
  could be dropped if rules_foreign_cc works around it for us.

- Preparing for specific corrections to get curl building curl.lib
  with cmake using clang-cl on Windows, starting at envoy and pushing
  the fix(es) to CMakeLists.txt upstream.

Risk Level: low
Testing: local
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

**Comment by htuch** [comment]
@wrowe this is failing `curl_test`.

**Comment by wrowe** [comment]
Thanks @htuch - that was a problem, we've had to disable UNIX_SOCKETS till we repair curl itself. Fixed the test short-term to not expect that feature on Windows builds, should be ready to merge once passing CI.

File: bazel/foreign_cc/curl-revert-cmake-minreqver.patch [hunk 1]
```diff
@@ -5,13 +5,13 @@ diff --git a/CMakeLists.txt b/CMakeLists.txt
...
-@@ -38,8 +38,7 @@
+@@ -37,8 +37,7 @@
  # To check:
  # (From Daniel Stenberg) The cmake build selected to run gcc with -fPIC on my box while the plain configure script did not.
  # (From Daniel Stenberg) The gcc command line use neither -g nor any -O options. As a developer, I also treasure our configure scripts's --enable-debug option that sets a long range of "picky" compiler options.
--cmake_minimum_required(VERSION 3.0...3.16 FATAL_ERROR)
+-cmake_minimum_required(VERSION 3.2...3.16 FATAL_ERROR)
 -
-+cmake_minimum_required(VERSION 3.0 FATAL_ERROR)
++cmake_minimum_required(VERSION 3.2 FATAL_ERROR)
```
Comment (moderation) [inline bazel/foreign_cc/curl-revert-cmake-minreqver.patch:14]: Is this a Windows specific patch? If so wondering if a comment should be added to https://github.com/envoyproxy/envoy/blob/master/bazel/repositories.bzl#L677-L678

Comment (htuch) [reply bazel/foreign_cc/curl-revert-cmake-minreqver.patch:14]: It would be good to understand why this is needed.

**Review by htuch — COMMENTED** [review]
LGTM modulo comment from @moderation

**Comment by sunjayBhatia** [comment]
/wait

reworking the patch

**Comment by wrowe** [comment]
We were able to drop the patch altogether by working around the CMake project's breakage. (This wasn't actually on the curl project's logic.)

**Comment by wrowe** [comment]
@htuch This should be much easier to follow now, are ready to review/merge

**Comment by moderation** [comment]
Really good to remove the patch @wrowe. Thanks. LGTM

**Review by htuch — APPROVED** [review]
LGTM, thanks for sorting out the patch situation!

**Pull request merged**